docs/drivers/generic.md

```diff
@@ -19,13 +19,48 @@ Machine to manage.
 The driver will perform a list of tasks on create:
 
 -   If docker is not running on the host, it will be installed automatically.
--   It will generate certificates to secure the docker daemon
+-   It will update the host packages (`apt-get update`, `yum update`...).
+-   It will generate certificates to secure the docker daemon.
 -   The docker daemon will be restarted, thus all running containers will be stopped.
+-   The hostname will be changed to fit the machine name.
 
+### Example
 
-    $ docker-machine create --driver generic --generic-ip-address=203.0.113.81 vm
+To create a machine instance, specify `--driver generic`, the IP address or DNS
+name of the host and the path to the SSH private key authorized to connect
+to the host.
 
-Options:
+    $ docker-machine create \
+      --driver generic \
+      --generic-ip-address=203.0.113.81 \
+      --generic-ssh-key=~/.ssh/id_rsa \
+      vm
+
+### Password-protected SSH keys
+
+When an SSH identity is not provided (with the `--generic-ssh-key` flag),
+the SSH agent (if running) will be consulted. This makes it possible to
+easily use password-protected SSH keys.
+
+Note that this usage is _only_ supported if you're using the external SSH client,
+which is the default behaviour when the `ssh` binary is available. If you're
+using the native client (with `--native-ssh`), using the SSH agent is not yet
+supported.
+
+    $ docker-machine create \
+      --driver generic \
+      --generic-ip-address=203.0.113.81 \
+      other
+
+### Sudo privileges
```

Comment:
This resolves / removes the need for #1586 

Thanks!

Reply:
This doesn't resolve the problem.

Reply:
Doesn't work for me either.